### PR TITLE
Add test paths filtering to diff tool

### PR DESCRIPTION
**Pull request by lukebjerring**
Allows specific path prefixes, which is useful for comparing results of 2 runs for a given folder.
e.g. `util/diff_runs.py --before=chrome@latest --after=safari@latest /css/css-images/`

### CODE REVIEW

File: util/diff_runs_test.py [hunk 1]
```diff
@@ -78,6 +80,28 @@ def results(sha, platform):
         self.assertIn('1 differences', logged)
         self.assertIn('1 tests', logged)
 
+    def test_cull_ignored_tests_dir(self):
```
Comment (mdittmer) [inline util/diff_runs_test.py]: Can we add tests for removes-all and removes-none?

Comment (lukebjerring) [reply util/diff_runs_test.py]: Removes none is done as 'test_no_difference', so just test_removes_all added.
